### PR TITLE
[FEATURE] add outline settings to font markers

### PR DESCRIPTION
**Pull request by nirvn**
This PR adds outline support for font markers. While most of this is UI logic, it did require the font marker's renderPoint() function to modified to use a QPainterPath's addText instead of a QPainter's drawText. 

@nyalldawson , your review on this is most needed. The travis build will fail on (at least) one font marker test as the mask file needs updating and I don't know how to do that.

Screenshot of the UI:
zzz

Screenshot of a not-so-ugly map featuring font marker outlines :smile::
bam
_The font marker's character is data-defined, using an expression to select ♀ when there are more female than male, and ♂ for the opposite scenario._

### CODE REVIEW

File: src/core/symbology-ng/qgsmarkersymbollayerv2.cpp [hunk 1]
```diff
@@ -2010,6 +2013,11 @@ QgsSymbolLayerV2* QgsFontMarkerSymbolLayerV2::create( const QgsStringMap& props
     angle = props["angle"].toDouble();
 
   QgsFontMarkerSymbolLayerV2* m = new QgsFontMarkerSymbolLayerV2( fontFamily, chr, pointSize, color, angle );
+
+  if ( props.contains( "color_border" ) )
```
Comment (nyalldawson) [inline src/core/symbology-ng/qgsmarkersymbollayerv2.cpp]: should use "outline_color" here - if we keep it consistent then the value is retained when switching between symbol layer types

File: src/core/symbology-ng/qgsmarkersymbollayerv2.cpp [hunk 2]
```diff
@@ -2152,20 +2173,63 @@ void QgsFontMarkerSymbolLayerV2::renderPoint( QPointF point, QgsSymbolV2RenderCo
   if ( !p )
     return;
 
-  QColor penColor = mColor;
+  QMatrix transform;
```
Comment (nyalldawson) [inline src/core/symbology-ng/qgsmarkersymbollayerv2.cpp]: QMatrix is obselete (see http://doc.qt.io/qt-4.8/qmatrix.html). Use QTransform instead.

File: src/gui/symbology-ng/qgssymbollayerv2widget.cpp [hunk 3]
```diff
@@ -2544,6 +2551,12 @@ void QgsFontMarkerSymbolLayerV2Widget::setSymbolLayer( QgsSymbolLayerV2* layer )
   btnColor->blockSignals( true );
   btnColor->setColor( mLayer->color() );
   btnColor->blockSignals( false );
+  btnBorderColor->blockSignals( true );
```
Comment (nyalldawson) [inline src/gui/symbology-ng/qgssymbollayerv2widget.cpp]: whileBlocking( btnBorderColor )->setColor( mLayer->outlineColor() );

(I realise there's a bunch of offending code around this, but may as well start somewhere!)

**Comment by nyalldawson** [comment]
@nirvn Looks good to me! Don't forget the dialog tab order too.

One question - what's the intended way to disable the outline? Set it to transparent? 0 width?

**Comment by nirvn** [comment]
@nyalldawson , thanks for the review, correcting stuff now. 

0 width = no outline (I use a label hint in the spin box for that).

File: src/core/symbology-ng/qgsmarkersymbollayerv2.cpp (hunk 1 above)
Comment (nirvn) [inline src/core/symbology-ng/qgsmarkersymbollayerv2.cpp]: @nyalldawson , I noticed that there seems to be a border vs outline "tension" throughout the symbology code. Have we settled on using outline for border/outline?

**Comment by nirvn** [comment]
@nyalldawson I've added a test for the outline feature. Since my dev machine is running on Ubuntu Wily, the test will probably fail as the font metrics are slightly different on this version. I'll make a mask as soon as cdash provides a rendered image.

**Comment by nyalldawson** [comment]
Great work!

**Comment by nirvn** [comment]
@nyalldawson thanks. Can you cherry-pick this commit ( https://github.com/nirvn/QGIS/commit/be28277d53a8c9834f04bdf4e786934d3863bf63 ) and apply it onto master? Sorry, I should have caught this missing bit yesterday.

**Pull request merged**